NDTensors/src/tupletools.jl

```diff
@@ -180,13 +180,13 @@ end
 
 Remove the value at pos and insert the elements in val
 """
-function insertat(t::NTuple{N}, val::NTuple{M}, pos::Integer) where {N,M}
+function insertat(t::Tuple, val::Tuple, pos::Integer)
+  N, M = length(t), length(val)
+  @assert pos <= N
```

Comment:
```suggestion
  @boundscheck pos in 1:length(t) || throw(BoundsError(t, pos))
```

Reply:
Nice to know about that & I agree it's better